docs/README.md

```diff
@@ -34,6 +34,55 @@ You'll need a `kubectl`-style config file to connect to your cluster.
 `$HOME/.kube/config`. You can specify an alternate kubeconfig connection file
 with `--kubeconfig`, or the env var `$KUBECONFIG`, for any command.
 
+## Kn Config
+
+To better customize Kn there is a small set of configuration parameters you are able to setup. In particular you can specify where your Kn plugins are located and how they are found. The Kn configuration file is meant to capture these limited configuration options. Let's explore this file, it's location, and the current options you are able to change with it.
+
+### Location
+
+By default the Kn config file is located in the user's home directory under `.kn` hidden subdirectory and is named `config.yaml`. So in `~/.kn/config.yaml`. It is not created for you by default. You can easily create one with:
+
+```bash
+mkdir ~/.kn
+touch ~/.kn/config.yaml
```

Comment:
Why would you ever do this ? Wouldn't you just create one with content when needed ?

We could also create one by default if not existing, with some commented out sample configuration. That is what many other tools do as well and is the best way to ramp up.

Reply:
Either way, if you do above and run Kn you will have the default settings. I was documenting current behavior. I am OK to auto create if not present everytime we run Kn.

Reply:
Resolving this. We can open an issue to add the default config as needed and add a line saying that the config file will be created. Either way this current behavior would be correct.